keps/sig-node/3619-supplemental-groups-policy/README.md

```diff
@@ -0,0 +1,837 @@
+
+However, Kubernetes provides no API surface to prevent this bypass, although it could sometimes lead to a security concern. Because the behavior is implemented in CRI implementations actually, To mitigate this, the cluster administrators will need to deploy a custom low-level container runtime(e.g., [pfnet-research/strict-supplementalgroups-container-runtime](https://github.com/pfnet-research/strict-supplementalgroups-container-runtime)) that modifies OCI container runtime spec(`config.json`) produced by CRI implementations (e.g., containerd, cri-o). A custom `RuntimeClass` would be introduced for it. Nevertheless, It would be an extra operational burden for cluster administrators.
+
+Thus, this KEP proposes to offer a new API field named `SupplementalGroupsPolicy` that customize to respect/ignore the group membership of the primary UID of the container defined in the container image when applying `SupplementalGroups` to the container processes. The new API allows cluster administrators to deploy security policies that protect the `SupplementalGroupsPolicy` field in the cluster to avoid the unexpected bypass of `SupplementalGroups` described above.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+To Provide a new API field to control how the groups of the first process in the container are calculated.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+N/A
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+A new field named `SupplementalGroupsPolicy` will be introduced to `PodSecurityContext`. Allowed values are:
+
+- `RespectGroupsInImage`(_default if not specified_): This policy indicates that it _respects_ group memberships defined in the container image for the container's primary UID in supplementary groups of the first container process. Its resultant supplementary groups are 
+  - the container's primary GID, 
+  - the `fsGroup` (if specified), 
+  - `supplementalGroups` (if specified), and 
+  - the groups defined in the container image for the container's primary UID.
+- `IgnoreGroupsInImage`: This policy indicates that it _ignores_ group memberships defined in the container image for the container's primary UID in supplementary groups of the first container process. Its resultant supplementary groups are
+  - the container's primary GID, 
+  - the `fsGroup` (if specified), and 
+  - `supplementalGroups` (if specified).
+
```

Comment:
Can we list two alternatives?
1. Implement this in container runtimes as an option. The con is that it isn't portable configuration.
2. Users could modify their images to control the supplemental groups. The con is that it is more work and users won't always have an option to do that.